src/main/java/de/dennisguse/opentracks/services/TrackDeleteService.java

```diff
@@ -137,35 +122,28 @@ public TrackDeleteService getService() {
     }
 
     public static class DeleteStatus {
-        private final int progress;
-        private final int max;
-        private final Track.Id trackId;
+        private final List<Track.Id> trackIds;
 
         /**
-         * @param trackId  Track.Id just deleted if any.
-         * @param progress number of deletes.
-         * @param max      total of deletes to be done.
+         * @param trackIds List of deleted Track.Ids.
          */
-        private DeleteStatus(@Nullable Track.Id trackId, int progress, int max) {
-            this.trackId = trackId;
-            this.progress = progress;
-            this.max = max;
+        private DeleteStatus(@Nullable List<Track.Id> trackIds) {
+            this.trackIds = trackIds;
         }
 
         public boolean isFinished() {
-            return progress == max;
+            return true;
         }
 
         public boolean isDeleted(Track.Id trackId) {
-            return this.trackId != null && this.trackId.equals(trackId);
+            return this.trackIds != null && this.trackIds.contains(trackId);
         }
 
         @NonNull
         @Override
         public String toString() {
             return "DeleteStatus{" +
-                    "number of deletes=" + progress +
-                    ", total=" + max +
+                    "trackIds=" + trackIds +
```

Comment:
This should not work as toString is not nicely implemented on ArrayList.

Reply:
This is properly implemented in a super class of ArrayList: https://cs.android.com/android/platform/superproject/+/master:libcore/ojluni/src/main/java/java/util/AbstractCollection.java;drc=12fae6f00fd514c510fa58f64c8870ab062733fd;l=464